transparent-shell@siroj42.github.io/extension.js

```diff
@@ -4,109 +4,74 @@ const ExtensionUtils = imports.misc.extensionUtils;
 
 const Extension = ExtensionUtils.getCurrentExtension();
 
-let gschema = Gio.SettingsSchemaSource.new_from_directory(
-    Extension.dir.get_child('schemas').get_path(),
-    Gio.SettingsSchemaSource.get_default(),
-    false
+const gschema = Gio.SettingsSchemaSource.new_from_directory(
+  Extension.dir.get_child("schemas").get_path(),
+  Gio.SettingsSchemaSource.get_default(),
+  false
 );
 
-let settings = new Gio.Settings({
-    settings_schema: gschema.lookup('org.gnome.shell.extensions.transparent-shell', true)
+const settings = new Gio.Settings({
+  settings_schema: gschema.lookup(
+    "org.gnome.shell.extensions.transparent-shell",
+    true
+  ),
 });
 
-let settingskeys = {
-    "top-panel": 0,
-    "dash": 0,
-    "search": 0,
-}
-
-
-function init(){
-    log("Starting transparent-shell extension");
-}
+const settingskeys = {
+  "top-panel": 0,
+  dash: 0,
+  search: 0,
+};
 
-function enable() {
-    Object.keys(settingskeys).forEach(key => {
-        if (settings.get_boolean(key)) {
-            enable_fkt(key);
-        }
-        settingskeys[key] = settings.connect(
-            'changed::'+key,
-            someChange
-        );
-    });
-}
+const SHELL_TRANSPARENCY = "shell-transparency";
 
-function disable() {
-    Object.keys(settingskeys).forEach(key => {
-        disable_fkt(key);
-        settings.disconnect(settingskeys[key]);
-    });
+function init() {
+  log("Starting transparent-shell extension");
 }
 
-
-
-function someChange(_,key){
-    if(settings.get_boolean(key)){
-        enable_fkt(key);
-    }else{
-        disable_fkt(key);
+const toggleTransparency = (el, enabled = true) =>
+  el &&
+  el[enabled ? "add_style_class_name" : "remove_style_class_name"](
+    SHELL_TRANSPARENCY
+  );
+
+const componentToggles = {
+  "top-panel": (isEnabled) => {
+    try {
+      [Main.panel, ...Main.mmPanel].forEach((mmpanel) => {
```

Comment:
When I run the extension on Gnome 42, this leads to an error:
```
TypeError: Main.mmPanel is undefined
```
Because of this, everything inside the forEach() is not even executed, and therefore, the panel transparency is broken. Could you please add some code to catch this error?

Reply:
Thanks for your comment! I didn't test that on anything other than gnome 42 and assumed the error would have been caught by the try/catch. I will fix it as soon as I can! :)

Reply:
I think I have fixed it now, can you please verify? It should work on Gnome 42 and below.

Reply:
@Siroj42 Hi! :) Just pining you, hope you have time to review the change. Thanks!